pkg/controller/resourceclaim/controller_test.go

```diff
@@ -508,11 +510,47 @@ func TestResourceClaimEventHandler(t *testing.T) {
 
 	var em numMetrics
 
+	expectQueue := func(tCtx ktesting.TContext, expectedKeys []string) {
+		g := gomega.NewWithT(tCtx)
+		tCtx.Helper()
+
+		lenDiffMessage := func() string {
+			actualKeys := []string{}
+			for ec.queue.Len() > 0 {
+				actual, _ := ec.queue.Get()
+				actualKeys = append(actualKeys, actual)
+				ec.queue.Forget(actual)
+				ec.queue.Done(actual)
+			}
+			return "Workqueue does not contain expected number of elements\n" +
+				"Diff of elements (- expected, + actual):\n" +
+				diff.Diff(expectedKeys, actualKeys)
+		}
+
+		g.Eventually(ec.queue.Len).
+			WithTimeout(5*time.Second).
+			Should(gomega.Equal(len(expectedKeys)), lenDiffMessage)
+		g.Consistently(ec.queue.Len).
+			WithTimeout(1*time.Second).
+			Should(gomega.Equal(len(expectedKeys)), lenDiffMessage)
+
+		for _, expected := range expectedKeys {
+			actual, shuttingDown := ec.queue.Get()
+			g.Expect(shuttingDown).To(gomega.BeFalseBecause("workqueue is unexpectedly shutting down"))
+			g.Expect(actual).To(gomega.Equal(expected))
+			ec.queue.Forget(actual)
+			ec.queue.Done(actual)
+		}
```

Comment:
This logic doesn't feel perfectly airtight and it slows the test way down (from ~3s to ~12s, mostly from the `Consistently`), so I'm open to ways to make that more robust and/or eliminate the long timeouts here. This might be a place that could make use of the [`Mock` workqueue](https://github.com/kubernetes/kubernetes/blob/6ed5b60f71930d51bfcf8bfa6f3506b099811318/staging/src/k8s.io/dynamic-resource-allocation/internal/workqueue/mockqueue.go#L46) if it were importable here.